game/src/edit/traffic_signals/picker.rs

```diff
@@ -48,27 +48,26 @@ impl State<App> for SignalPicker {
         ctx.canvas_movement();
         if ctx.redo_mouseover() {
             app.primary.current_selection = app.mouseover_unzoomed_roads_and_intersections(ctx);
+            app.select_only_intersections(|_| true);
```

Comment:
It's a little surprising to me how this works - we explicitly set current_selection, and then maybe immediately unset it in this method.

Do you think something like this would be viable?

Introduce a `app.mouseover_unzoomed_intersections` which only returns intersections:
```diff
- app.primary.current_selection = app.mouseover_unzoomed_roads_and_intersections(ctx);
- app.select_only_intersections(|_| true);
+ app.primary.current_selection = app.mouseover_unzoomed_intersections(ctx);
```

Use chain Option::filter upon assignment, rather than assigning and then potentially unsetting.
```diff
- app.primary.current_selection = app.mouseover_unzoomed_roads_and_intersections(ctx);
- app.select_only_intersections(|i| self.members.contains(&i));
+ app.primary.current_selection = app.mouseover_unzoomed_intersections(ctx).filter(|i| self.members.contains(i))
```

It's less "action at a distance" to my eye.

Reply:
Thanks, I like this approach much better! The only slight hiccup is that `mouseover_unzoomed_intersections` still returns `Option<ID>` to fit in, so we have to sneak in a typecast that we know won't fail: `app.primary.current_selection = app.mouseover_unzoomed_intersections(ctx).filter(|id| app.primary.map.maybe_get_traffic_signal(id.as_intersection()).is_some());`

An alternative I'm still open to is having `mouseover_unzoomed_intersections` return `Option<IntersectionID>` and adding an `.into()` to transform that to `Option<ID>`. Merging the first approach for now, but still thinking about it.

Reply:
> An alternative I'm still open to is having mouseover_unzoomed_intersections return Option<IntersectionID>

That makes sense to me - since we this won't be any other type of `ID`.  If a callsite needs `ID`, we could do the type erasure there.